### PR TITLE
feat: add mStable action contracts and tests

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

File: contracts/interfaces/mstable/IBoostedVaultWithLockup.sol [hunk 1]
```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.7.6;
+
+interface IBoostedVaultWithLockup {
+
+    function rawBalanceOf(address) external view returns (uint256);
+
+    /**
```
Comment (yippee-ki-yay) [inline contracts/interfaces/mstable/IBoostedVaultWithLockup.sol:8]: Remove /** */ comments as they sometimes break the flattener

File: contracts/views/MStableView.sol [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    function earned(address _vault, address _account)
+    public view returns (
+        uint256 amount
+    ) {
+        return IBoostedVaultWithLockup(_vault).earned(_account);
+    }
+
+    function pokeBoost(address _vault, address _account) external {
```
Comment (yippee-ki-yay) [inline contracts/views/MStableView.sol:44]: Why add a state changing func. in a view contract?

File: contracts/actions/mstable/MStableDeposit.sol [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            _params.amount = _params.bAsset.getBalance(_params.from);
+        }
+        _params.bAsset.pullTokensIfNeeded(_params.from, _params.amount);
+        _params.bAsset.approveToken(_params.mAsset, _params.amount);
+
+        uint256 mAssetsMinted = ImAsset(_params.mAsset).mint(_params.bAsset, _params.amount, _params.minOut, address(this));
+
+        _params.mAsset.approveToken(_params.saveAddress, mAssetsMinted);
+        if (_params.stake) {
+            credits = ISavingsContractV2(_params.saveAddress).depositSavings(mAssetsMinted, address(this));
```
Comment (yippee-ki-yay) [inline contracts/actions/mstable/MStableDeposit.sol:66]: credits = ISavingsContractV2(_params.saveAddress).depositSavings(mAssetsMinted, stake ? address(this) : _params.to);

Can pull this outside the if statement as the same operation is done in both cases just a diff. input.

File: contracts/actions/mstable/MStableClaim.sol [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import "../ActionBase.sol";
+
+contract MStableClaim is ActionBase, MainnetMStableAddresses {
+    using TokenUtils for address;
+    using SafeMath for uint256;
+
+    struct Params {
+        address vaultAddress;   // vault contract address for the imAsset (imAssetVault address)
+        address to;             // address that will receive the claimed MTA rewards
+        uint256 first;
```
Comment (yippee-ki-yay) [inline contracts/actions/mstable/MStableClaim.sol:18]: What are first and last? Maybe comments and/or better names

**Review by yippee-ki-yay — COMMENTED** [review]
Looking pretty clean, besides the few details I mentioned, once fixed should be able to merge.

File: contracts/actions/mstable/MStableDeposit.sol (hunk 3 above)
Comment (kjaredic) [reply contracts/actions/mstable/MStableDeposit.sol:66]: I think readability is better this way.

File: contracts/actions/mstable/MStableClaim.sol (hunk 4 above)
Comment (kjaredic) [reply contracts/actions/mstable/MStableClaim.sol:18]: First and last index in the users rewards array on which to claim.
I propose we document this in gitbook similarly how we did it for  firstHint, lastHint (Liquity).

**Review by yippee-ki-yay — APPROVED** [review]

**Pull request merged**